### PR TITLE
Add the initial package structure

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

**Comment by arcondello** [comment]
One question off the bat, should this be `dwave-scikit-learn-plugin` rather than `dwave-sklearn-plugin`? In scikit-learn they
```python
import sklearn
```
but
```bash
pip install scikit-learn
```

I feel like `dwave-scikit-learn-plugin` is over-verbose but perhaps consistency is more important here.

**Comment by arcondello** [comment]
Having now typed it a few times, I am warming to `dwave-scikit-learn-plugin`. It's only a few characters longer. But could easily go either way.

**Comment by juansebastianl** [comment]
Because of the import from `scikit-learn` being `import sklearn` I prefer our import being `import dwave-sklearn-plugin`

**Comment by arcondello** [comment]
To be clear, that will be the import in either case. I am asking whether we want to go with:

#### Option A
##### Install
```bash
pip install dwave-sklearn-plugin
```
##### Code
```python
from dwave.plugins.sklearn import ...
```

#### Option B
##### Install
```bash
pip install dwave-scikit-learn-plugin
```
##### Code
```python
from dwave.plugins.sklearn import ...
```

#### Option C
Something else?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,240 @@
+version: 2.1
```
Comment (randomir) [inline .circleci/config.yml:1]: Is this `config.yml` from dwave-hybrid? I'd like to update build/test jobs in orb-ocean accordingly, and simplify CI configs in ocean packages. What do you think?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    License :: OSI Approved :: Apache Software License
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3 :: Only
+license = Apache 2.0
+long_description = file: README.md
+name = dwave-sklearn-plugin
+project_urls =
+    Changes = https://github.com/dwavesystems/dwave-sklearn-plugin/releases
+    Documentation = https://docs.ocean.dwavesys.com
+    Souce Code = https://github.com/dwavesystems/dwave-sklearn-plugin
```
Comment (randomir) [inline setup.cfg:12]: typo

**Review by randomir — APPROVED** [review]
lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (arcondello) [reply .circleci/config.yml:1]: Yup, and works for me

**Comment by randomir** [comment]
Oh, and I prefer option B, `dwave-scikit-learn-plugin`.

File: setup.cfg (hunk 2 above)
Comment (arcondello) [reply setup.cfg:12]: Probably should be but it actually seems to work fine. I copied the format from [dwave-samplers](https://github.com/dwavesystems/dwave-samplers/blob/8a3122e1bde8eeb3505b1e4b8a337d66f9173b55/setup.cfg#L15) and it renders correctly on [PyPI](https://pypi.org/project/dwave-samplers/).

**Review by juansebastianl — APPROVED** [review]

**Comment by juansebastianl** [comment]
Okay, now that I've understood the import vs install I like option B as well, it seems consistent with the sci-kit paradigm

File: setup.cfg (hunk 2 above)
Comment (randomir) [reply setup.cfg:12]: `Souce Code` renders correctly?

Comment (arcondello) [reply setup.cfg:12]: It does indeed

Comment (randomir) [reply setup.cfg:12]: Ok, maybe diff will clarify it :smile: 
```suggestion
    Source Code = https://github.com/dwavesystems/dwave-sklearn-plugin
```

Comment (arcondello) [reply setup.cfg:12]: Ha! I am clearly confused. Will fix. Thanks

**Pull request merged**